docs/lib/datastore/getting-started.md

```diff
@@ -9,44 +9,155 @@ Amplify DataStore provides a programming model for leveraging shared and distrib
+    `Yes`
+? Select the default resolution strategy
+    `Auto Merge`
+? Do you have an annotated GraphQL schema?
+    `No`
+? Do you want a guided schema creation?
+    `No`
+? Provide a custom type name
+    `Post`
+```
+
+<amplify-callout warning>
+
+**Troubleshooting:** without the **conflict detection** configuration cloud sync will fail. In that case use `amplify update api` and choose **Enable DataStore for entire API** (this option will enable the conflict detection as described above).
+
+</amplify-callout>
+
+## Idiomatic persistence
+
+DataStore relies on platform standard data structures to represent the data schema in an idiomatic way. The persistence language is composed by data types that satisfies the `Model` interface and operations defined by common verbs such as `save`, `query` and `delete`.
+
+### Data schema
+
+The first step to create an app backed by a persistent datastore is to **define a schema**. DataStore uses GraphQL schema files as the definition of the application data model. The schema contains data types and relationships that represent the app's functionality.
 
 ### Sample schema
 
-For the purpose of this guide, let's use the following simple GraphQL schema:
+For the next steps, let's start with a schema for a small blog application. It has a single model, a `Post`. New types and constructs will be added to this base schema as more concepts are presented.
 
-```graphql
-enum PostStatus {
-  ACTIVE
-  INACTIVE
-}
+Open the `schema.graphql` file located by default at `amplify/backend/{api_name}/` and **define the model** `Post` as follows.
 
+```graphql
 type Post @model {
   id: ID!
   title: String!
-  rating: Int!
-  status: PostStatus!
+  rating: Int
+  content: String
 }
 ```
 
-<inline-fragment platform="js" src="~/lib/datastore/fragments/js/getting-started/generate-models.md"></inline-fragment>
-<inline-fragment platform="ios" src="~/lib/datastore/fragments/ios/getting-started/generate-models.md"></inline-fragment>
-<inline-fragment platform="android" src="~/lib/datastore/fragments/android/getting-started/generate-models.md"></inline-fragment>
+Now you will to convert the platform-agnostic `schema.graphql` into platform-specific data structures that will enforce the idiomatic persistence concepts. DataStore relies on code generation to guarantee schemas are correctly converted to platform code.
+
+Like the initial setup, models can be generated either using the IDE integration or Amplify CLI directly.
+
+### Code generation: Platform integration
+
+<inline-fragment platform="js" src="~/lib/datastore/fragments/js/getting-started/40_codegen.md"></inline-fragment>
+<inline-fragment platform="ios" src="~/lib/datastore/fragments/ios/getting-started/40_codegen.md"></inline-fragment>
+<inline-fragment platform="android" src="~/lib/datastore/fragments/android/getting-started/40_codegen.md"></inline-fragment>
+
+### Code generation: Amplify CLI
+
+Models can also be generated using the Amplify CLI directly.
```

Comment:
Suggestion, i know this makes it harder since we have to split up each platform, but i think it makes it more clear:

If you have chosen to not use AmplifyTools or want to generate models without issuing a build in XCode, you can use the following steps:

Reply:
I guess we are leaving this to do for another time...